debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerChangeTablePointer.java

```diff
@@ -47,16 +48,26 @@
     private ResultSetMapper<Object[]> resultSetMapper;
     private final ResultSet resultSet;
     private final int columnDataOffset;
+    private final SourceTimestampMode sourceTimestampMode;
 
-    public SqlServerChangeTablePointer(SqlServerChangeTable changeTable, ResultSet resultSet) {
-        this(changeTable, resultSet, COL_DATA);
+    public SqlServerChangeTablePointer(SqlServerChangeTable changeTable, ResultSet resultSet, SourceTimestampMode sourceTimestampMode) {
+        this(changeTable, resultSet, sourceTimestampMode, COL_DATA);
     }
 
-    public SqlServerChangeTablePointer(SqlServerChangeTable changeTable, ResultSet resultSet, int columnDataOffset) {
+    public SqlServerChangeTablePointer(SqlServerChangeTable changeTable, ResultSet resultSet, SourceTimestampMode sourceTimestampMode, int columnDataOffset) {
         super(changeTable, resultSet, columnDataOffset);
         // Store references to these because we can't get them from our superclass
         this.resultSet = resultSet;
         this.columnDataOffset = columnDataOffset;
+        this.sourceTimestampMode = sourceTimestampMode;
+    }
+
+    public Timestamp getCommitTimestamp() throws SQLException {
+        if (SourceTimestampMode.PROCESSING.equals(sourceTimestampMode)) {
+            return null;
+        }
+
+        return resultSet.getTimestamp(resultSet.getMetaData().getColumnCount());
```

Comment:
This might need to be optimized in order to not call `resultSet.getMetaData().getColumnCount()` on each row.

Reply:
I don't think it needs any optimization as it's highly unlikely for a method that basically returns the length of the columns array to become a performance bottleneck.